ros_buildfarm/status_page.py

```diff
@@ -39,6 +39,111 @@
 from .templates import get_template_path
 
 
+def build_blocked_source_entries_page(
+        config_url, rosdistro_name,
+        output_dir, copy_resources=False):
+    from rosdistro import get_cached_distribution
+    from rosdistro import get_index
+    from rosdistro.dependency_walker import DependencyWalker
+
+    start_time = time.localtime()
+
+    config = get_config_index(config_url)
+    index = get_index(config.rosdistro_index_url)
+
+    rosdistro_name = _rosdistro_or_latest(index, rosdistro_name)
+    print('Getting blocked source entries for', rosdistro_name)
+
+    try:
+        prev_rosdistro_name = _prev_rosdistro(index, rosdistro_name)
+    except ValueError as e:
+        print(e.args[0], file=sys.stderr)
+        exit(-1)
+
+    print('Comparing', rosdistro_name, 'with', prev_rosdistro_name)
+
+    dist = get_cached_distribution(index, rosdistro_name)
+    prev_dist = get_cached_distribution(index, prev_rosdistro_name)
+
+    prev_walker = DependencyWalker(prev_dist)
+
+    prev_released_repos = set(_released_repos(prev_dist))
+    source_entry_repos = set(_source_entry_repos(dist))
+    missing_repos = prev_released_repos.difference(source_entry_repos)
+
+    # Assume repos will provide the same packages as previous distro
+    missing_packages = set(_released_packages(prev_dist, missing_repos))
+
+    repos_info = defaultdict(dict)
+
+    # Give all repos some basic info
+    for repo_name in prev_released_repos.union(source_entry_repos).union(missing_repos):
+        repos_info[repo_name]['url'] = ''
+        repos_info[repo_name]['repos_blocking'] = set()
+        repos_info[repo_name]['recursive_repos_blocking'] = set()
+        repos_info[repo_name]['released'] = False
+        repos_info[repo_name]['version'] = 'no'
+        repos_info[repo_name]['repos_blocked_by'] = {}
+        repos_info[repo_name]['maintainers'] = defaultdict(dict)
+
+    for repo_name in prev_released_repos:
+        repos_info[repo_name]['url'] = _repo_url(prev_dist, repo_name)
+
+    # has a source entry? Lets call that 'released' with a 'version' to reuse
+    # logic for the blocked releases page
```

Comment:
The naming is really awkward. Can you point to what exact parts are being reused to get an idea if it is worth the weird naming?

Reply:
I think it's just reusing [`_format_repo_table_row`](https://github.com/ros-infrastructure/ros_buildfarm/blob/467c34fb812281c32307c6b6339c2da13d394907/ros_buildfarm/status_page.py#L644)

Reply:
Since `_format_repo_table_row` is fairly long I am fine with keeping it reusing that. Maybe update the comment to mention that function / line explicitly.

Reply:
Updated comment to mention `_format_repo_table_row` in 6667374